code/game/objects/effects/particle_holder.dm

```diff
@@ -57,5 +65,35 @@
 
 /// Sets the particles position to the passed coordinate list (X, Y, Z)
 /// See [https://www.byond.com/docs/ref/#/{notes}/particles] for position documentation
-/obj/effect/abstract/particle_holder/proc/set_particle_position(list/pos)
+/obj/effect/abstract/particle_holder/solo/proc/set_particle_position(list/pos)
 	particles.position = pos
+
+///Stores an index of particles and their instances for reusing : list(particle_type = list(INSTANCES BABEEYYYY))
+GLOBAL_LIST_EMPTY(shared_particle_holders)
+
+///Give a particle effect to an atom from a global list. Use this if you have a lot of particle holders on someones
+///screen and client performance is a problem. This way it's only calculated x amount of times and you can replicate it without putting too much extra burden on clients
+///particle_flags : holder behaviour. holder_type: defines whether we are shared or use our own. pool_size: if shared, how many do we share? (only first pool_size counts)
+/proc/apply_particles_to(atom/particlee, particle_type, particle_flags = NONE, holder_type = PARTICLES_SINGULAR, pool_size)
+	switch(holder_type) //yes this definitely asuredly needs to be a switch because uhhhhhhhhh looks nice
+		if(PARTICLES_SINGULAR)
+			return new /obj/effect/abstract/particle_holder/solo (particlee, particle_type, particle_flags)
+
+		if(PARTICLES_SHARED)
+			var/list/effect_pool = GLOB.shared_particle_holders[particle_type]
+			if(!LAZYLEN(effect_pool)) //no particles yet so make a pool of instances to pick from
+				for(var/i in 1 to pool_size)
+					effect_pool += new /obj/effect/abstract/particle_holder/shared(null, particle_type)
+
+				var/obj/effect/abstract/particle_holder/shared/holder_instance = pick(effect_pool)
+				holder_instance.apply_particles_to(particlee)
+				return holder_instance
+
+///Many particle slow? Have one or two particle and copy paste everywhere. Similair to particle_holder but meant for having multiple vis_locs
+///If using: make a subtype with a set particle and pool_size for the amount of instances you want to randomly pick from
+/obj/effect/abstract/particle_holder/shared
+
+///Apply the particle effect to x obj. Turfs are also fine ignore the typecasting HAHAHHAHAHAHAH i hate byond
+/obj/effect/abstract/particle_holder/shared/proc/apply_particles_to(atom/movable/add_to)
+	add_to.vis_contents += src
```

Comment:
did you want to support the particle flags here? I was thinking that was the upside of the shared parent